src/node_buffer.cc

```diff
@@ -741,6 +749,37 @@ void SlowByteLengthUtf8(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(args[0].As<String>()->Utf8Length(env->isolate()));
 }
 
+#if NODE_HAS_SIMD_NEON
+uint32_t FastByteLengthUtf8(Local<Value> receiver,
+                            const v8::FastOneByteString& source) {
+  const auto data = reinterpret_cast<const uint8_t*>(source.data);
+  const uint8x16_t mask = vdupq_n_u8(0x80);
+  uint8x16_t result_vector = vdupq_n_u8(0);
+  size_t i = 0;
+
+  for (; i < source.length; i += 16) {
+    // load 16 bytes from data
+    uint8x16_t values = vld1q_u8(data + i);
+
+    // extract the high bits using 0x80 mask
+    uint8x16_t high_bits = vcgeq_u8(values, mask);
```

Comment:
`vcgeq_u8` will put `0xff` or `0x00` into a lane, not `0x1` or `0x0`. You probably want `vshrq_n_u8`. As-is this function doesn't return correct results, e.g. this test case fails once the optimizer kicks in:
```console
$ out/Release/node -p 'while(Buffer.byteLength("ÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿ")==32);'
```

Reply:
You're right. Updated.